scripts/segmentation/test.py

```diff
@@ -207,37 +217,87 @@ def benchmarking(model, args):
-        model = get_segmentation_model(model=args.model, dataset=args.dataset, ctx=args.ctx,
-                                       backbone=args.backbone, norm_layer=args.norm_layer,
-                                       norm_kwargs=args.norm_kwargs, aux=args.aux,
-                                       base_size=args.base_size, crop_size=args.crop_size)
-        # load local pretrained weight
-        assert args.resume is not None, '=> Please provide the checkpoint using --resume'
-        if os.path.isfile(args.resume):
-            model.load_parameters(args.resume, ctx=args.ctx)
+    if not args.deploy:
+        if args.calibration:
+            args.pretrained = True
+        # create network
+        if args.pretrained:
+            model = get_model(model_prefix, pretrained=True)
+            model.collect_params().reset_ctx(ctx=args.ctx)
+        else:
+            assert "_in8" not in model_prefix, "Currently, Int8 models are not supported when pretrained=False"
+            model = get_segmentation_model(model=args.model, dataset=args.dataset, ctx=args.ctx,
+                                           backbone=args.backbone, norm_layer=args.norm_layer,
+                                           norm_kwargs=args.norm_kwargs, aux=args.aux,
+                                           base_size=args.base_size, crop_size=args.crop_size)
+            # load local pretrained weight
+            assert args.resume is not None, '=> Please provide the checkpoint using --resume'
+            if os.path.isfile(args.resume):
+                model.load_parameters(args.resume, ctx=args.ctx)
+            else:
+                raise RuntimeError("=> no checkpoint found at '{}'" \
+                    .format(args.resume))
+        if args.quantized:
+            model.hybridize(static_alloc=True, static_shape=True)
         else:
-            raise RuntimeError("=> no checkpoint found at '{}'" \
-                .format(args.resume))
+            model.hybridize()
+    else:
+        model_prefix = 'deploy_int8' if args.quantized else 'deploy'
+        model = mx.gluon.SymbolBlock.imports('{}-symbol.json'.format(args.model_prefix),
+              ['data'], '{}-0000.params'.format(args.model_prefix))
+        model.hybridize(static_alloc=True, static_shape=True)
 
     print("Successfully loaded %s model" % model_prefix)
     print('Testing model: ', args.resume)
+
+    # benchmark
+    if args.benchmark:
+        print('------benchmarking on %s model------' % model_prefix)
+        benchmarking(model, args)
+        sys.exit()
+
     # image transform
     input_transform = transforms.Compose([
         transforms.ToTensor(),
         transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
     ])
 
-    if not args.benchmark:
-        if '_int8' in model_prefix:
-            test_quantization(model, args, input_transform)
+    if args.calibration or '_int8' in model_prefix:
```

Comment:
@xinyu-intel I'm not sure if it is appropriate to put `calibration` part into `test_quantization` function. Then, changes for common part will become much smaller (only have changes in `test_quantization` function).